src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Constants/ErrorMessages.cs

```diff
@@ -10,5 +12,9 @@ internal class ErrorMessages
         public const string FeatureFlagInvalidJsonProperty = "Invalid property '{0}' for feature flag. Key: '{1}'. Found type: '{2}'. Expected type: '{3}'.";
         public const string FeatureFlagInvalidFormat = "Invalid json format for feature flag. Key: '{0}'.";
         public const string InvalidKeyVaultReference = "Invalid Key Vault reference.";
+        public const string SnapshotReferenceInvalidFormat = "Invalid snapshot reference format for key '{0}' (label: '{1}').";
+        public const string SnapshotReferenceInvalidJsonProperty = "Invalid snapshot reference format for key '{0}' (label: '{1}'). The '" + JsonFields.SnapshotName + "' property must be a string value, but found {2}.";
```

Comment:
Can you pass the value of `JsonFields.SnapshotName` to the error message in SnapshotReferenceParser instead of adding this dependency in ErrorMessages.